File: Python/ipbin.py
#!/usr/bin/env python3
import sys


def transIP(decIp):
    segs = decIp.split('.')
    bins = [bin(int(x, 10))[2:].zfill(8) for x in segs]
    return ".".join(bins)


if __name__ == "__main__":
    for arg in sys.argv[1:]:
        print(transIP(arg))

File: Rust/ipbin.rs
use std::env;

fn trans_ip(ip: &str) -> String {
    ip.split('.')
        .map(|seg| match seg.parse::<u8>() {
            Ok(i) => format!("{:08b}", i),
            Err(_) => "{ERR}".to_string(),
        })
        .collect::<Vec<String>>()
        .join(".")
}

fn main() {
    for arg in env::args().skip(1) {
        println!("{}", trans_ip(&arg));
    }
}
